lib/modules/pipeline/index.js

```diff
@@ -1,64 +1,60 @@
-const fs = require('../core/fs.js');
+const fs = require('../../core/fs.js');
 const async = require('async');
-const ProcessLauncher = require('../core/processes/processLauncher');
-const utils = require('../utils/utils.js');
-const constants = require('../constants');
+const utils = require('../../utils/utils.js');
+const ProcessLauncher = require('../../core/processes/processLauncher');
+const constants = require('../../constants');
 
 class Pipeline {
-  constructor(options) {
-    this.env = options.env;
-    this.buildDir = options.buildDir;
-    this.contractsFiles = options.contractsFiles;
-    this.assetFiles = options.assetFiles;
-    this.events = options.events;
-    this.logger = options.logger;
-    this.plugins = options.plugins;
+  constructor(embark, options) {
+    this.env = embark.config.env;
+    this.buildDir  = embark.config.buildDir;
+    this.contractsFiles = embark.config.contractsFiles;
+    this.assetFiles = embark.config.assetFiles;
+    this.events = embark.events;
+    this.logger = embark.config.logger;
+    this.plugins = embark.config.plugins;
     this.webpackConfigName = options.webpackConfigName;
     this.pipelinePlugins = this.plugins.getPluginsFor('pipeline');
     this.isFirstBuild = true;
 
-    fs.removeSync(this.buildDir);
+    this.events.setCommandHandler('pipeline:build', (options, callback) => this.build(options, callback))
```

Comment:
Just an aesthetic suggestion. You can do `this.events.setCommandHandler('pipeline:build', this.build.bind(this))`

Reply:
Yes! I considered that too, but then I my brain found the fat arrow version easier to parse. :)